app/src/main/java/com/tomtom/sdk/examples/maps/ConfigurableMapActivity.kt

```diff
@@ -0,0 +1,198 @@
+package com.tomtom.sdk.examples.maps
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.tomtom.sdk.examples.BuildConfig
+import com.tomtom.sdk.examples.R
+import com.tomtom.sdk.examples.databinding.ActivityMapViewBinding
+import com.tomtom.sdk.location.GeoPoint
+import com.tomtom.sdk.location.LocationProvider
+import com.tomtom.sdk.location.OnLocationUpdateListener
+import com.tomtom.sdk.location.android.AndroidLocationProvider
+import com.tomtom.sdk.map.display.MapOptions
+import com.tomtom.sdk.map.display.TomTomMap
+import com.tomtom.sdk.map.display.camera.CameraOptions
+import com.tomtom.sdk.map.display.location.LocationMarkerOptions
+import com.tomtom.sdk.map.display.ui.MapFragment
+import com.tomtom.sdk.map.display.ui.MapView
+
+/**
+ * This activity is responsible for displaying the TomTom Vector Map, both with location and without.
+ */
+class ConfigurableMapActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMapViewBinding
+    private lateinit var mapView: MapView
+    private lateinit var mapFragment: MapFragment
+    private lateinit var tomTomMap: TomTomMap
+    private lateinit var locationProvider: LocationProvider
+    private lateinit var onLocationUpdateListener: OnLocationUpdateListener
+    private lateinit var goBackImageButtonView: ImageView
+
+    companion object {
+        const val AMSTERDAM_GEO_POINT_LATITUDE = 52.379189
+        const val AMSTERDAM_GEO_POINT_LONGITUDE = 4.899431
+        const val CAMERA_ZOOM_VALUE = 8.0
+    }
+
+    /**
+     * Navigation SDK is only available upon request.
+     * Use the API key provided by TomTom to start using the SDK.
+     */
+    private val apiKey = BuildConfig.TOMTOM_API_KEY
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMapViewBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+        loadMapViewPage()
+
+        /**
+         * Triggered on button go back click to return to the main view with the map styles
+         */
+        goBackImageButtonView.setOnClickListener {
+            onBackPressed()
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView = mapFragment.view as MapView
+        mapView.contentDescription = applicationContext.resources.getString(R.string.map_not_ready)
+    }
+
+    /**
+     * All functions needed to load the activity_map_view layout page
+     */
+    private fun loadMapViewPage() {
+        initMap()
+        initializeUIElements()
+        setTagsToViewsWithDrawable()
+    }
+
+    private fun initializeUIElements() {
+        goBackImageButtonView = binding.goBackImageButton
```

Comment:
This property is only used twice, I don't think it is needed since we are using view binding